### ISSUE TITLE
Session support?

### ISSUE THREAD

**Issue by M-Ahall** [amitbet/vncproxy#26]
I'm trying to get vncproxy running to connect both clients and servers. Is the session support implemented to allow more than one target server?

**Comment by BirknerAlex** [comment]
I'm also struggling around with this "issue". Currently there are 2 issues with multi-client-server support.

1. The proxy structs `sessionManager   *SessionManager` value isn't publicly available and so there is no way to access it from outside to add/remove sessions. 

I've patched the struct and changed the `sessionManager` variable to `SessionManager`.

```go

	a.SessionManager = &proxy.SessionManager{}
	a.SessionManager.Init()

	a.VncProxy = &proxy.VncProxy{
		WsListeningURL: "http://127.0.0.1:5900/",
		UsingSessions:  true,
		SessionManager: a.SessionManager,
	}
```

2. The WS handler ignores the sessionId provided in the url so far I can tell but from code [perspective](https://github.com/amitbet/vncproxy/blob/master/server/ws-server-go.go#L31) it should work 

I expected when connecting via WS to `ws://127.0.0.1:5900/<sessionid>` that it works, but sessionId is always `""`.

--- TLDR:

What I want to to, I am running several servers with VNC enabled there the IP is behind a private network, I want to place a proxy service which validates the connection (JWT token), checks against the database if the user is allowed to connect and if yes, I want to forward the connection to the right service behind including recording the session. 

So the session should be dynamically added (on connect) and removed once the JWT token is expired or by other reasons.

Another thing I hoped from this project is multiplexing the server connection, in my case the VNC servers just allowing one client per server at the same time, I hoped this proxy just creates one server connection and allows me multiple users connecting to the same upstream at the same.
